### DISCUSSION TITLE
Questions about installing mxnet and gluoncv on nvidia jetson tx2

### DISCUSSION THREAD

**Discussion by Jessespace** [apache/incubator-mxnet#14953] [Q&A] [answered]
1. I tried the mxnet official website to use the source installation steps, failed after many attempts.
2. Use sudo pip install mxnet-jetson to install successfully, but the latest version is 1.2.1, and gluoncv requires mxnet version to be at least 1.3.0 (otherwise when importing gluoncv error)
How do I upgrade the mxnet version on jetson to 1.3.0 or higher?

[a comment by an automated account is not shown]

**Comment by vdantu** [comment]
@mxnet-label-bot add [question, build]
@Jessespace : Could you please share the errors you faced when trying to compile from source? 
@KellenSunderland : Would you be able to provide a context on mxnet-jetson package and its upgrade time frames?

**Comment by Jessespace** [comment]
> @mxnet-label-bot add [question, build]
> @Jessespace : Could you please share the errors you faced when trying to compile from source?
> @KellenSunderland : Would you be able to provide a context on mxnet-jetson package and its upgrade time frames?

My device is **nvidia jetson tx2**
1. I followed the steps of [the official website](http://mxnet.incubator.apache.org/versions/master/install/index.html?platform=Devices&language=Python&processor=CPU), failed. ( Sorry, I didn't record the errors. )
2. I referenced [here](https://devtalk.nvidia.com/default/topic/1049293/jetson-nano/i-was-unable-to-compile-and-install-mxnet-on-the-jetson-nano-is-there-an-official-installation-tutorial-/1https://devtalk.nvidia.com/default/topic/1049293/jetson-nano/i-was-unable-to-compile-and-install-mxnet-on-the-jetson-nano-is-there-an-official-installation-tutorial-/1) to recompile and succeeded. This is the follow steps:
```python
sudo apt-get update
sudo apt-get -y install git build-essential libatlas-base-dev libopencv-dev graphviz python-pip
sudo pip install pip --upgrade
sudo pip install setuptools numpy --upgrade
sudo pip install graphviz jupyter

git clone https://github.com/apache/incubator-mxnet.git --recursive
cd incubator-mxnet

cp make/config.mk .
sed -i 's/USE_CUDA = 0/USE_CUDA = 1/' config.mk
sed -i 's/USE_CUDA_PATH = NONE/USE_CUDA_PATH = \/usr\/local\/cuda/' config.mk
sed -i 's/USE_CUDNN = 0/USE_CUDNN = 1/' config.mk
sed -i '/USE_CUDNN/a CUDA_ARCH := -gencode arch=compute_62,code=sm_62' config.mk

Editing 3rdparty/mshadow/make/mshadow.mk and adding the following after line 122:
MSHADOW_CFLAGS += -DMSHADOW_USE_PASCAL=1

make -j $(nproc)

cd ..     # (incubator-mxnet dir)
export MXNET_HOME=$(pwd)
echo "export PYTHONPATH=$MXNET_HOME/python:$PYTHONPATH" >> ~/.bashrc
source ~/.bashrc
```
The main difference between the above steps and the official website tutorial is that:
`cp make/config.mk`  `cp make/crosscompile.jetson.mk config.mk` ,What is the difference between the two?
In addition, can you provide **compiled mxnet1.4.0** ? The compilation process is too long, Thanks.

**Comment by aaronmarkham** [comment]
@Jessespace Can you try the instructions here: https://github.com/apache/incubator-mxnet/blame/master/docs/install/install-jetson.md

**Comment by haojin2** [comment, marked as answer]
@Jessespace I've tried installation on Jetson TX2 before and I would definitely recommend using the cross-compilation method.
